grafana/resource_role_assignment.go

```diff
@@ -0,0 +1,145 @@
+* [Official documentation](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/developers/http_api/access_control/)
+`,
+		CreateContext: UpdateRoleAssignments,
+		UpdateContext: UpdateRoleAssignments,
+		ReadContext:   ReadRoleAssignments,
+		DeleteContext: UpdateRoleAssignments,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"role_uid": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Grafana RBAC role UID.",
+			},
+			"users": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "IDs of users that the role should be assigned to.",
+				Elem: &schema.Schema{
+					Type: schema.TypeInt,
+				},
+			},
+			"teams": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "IDs of teams that the role should be assigned to.",
+				Elem: &schema.Schema{
+					Type: schema.TypeInt,
+				},
+			},
+			"service_accounts": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "IDs of service accounts that the role should be assigned to.",
+				Elem: &schema.Schema{
+					Type: schema.TypeInt,
+				},
+			},
+		},
+	}
+}
+
+func ReadRoleAssignments(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+	uid := d.Id()
+	assignments, err := client.GetRoleAssignments(uid)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := setRoleAssignments(assignments, d); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func UpdateRoleAssignments(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if !d.IsNewResource() && !d.HasChange("users") && !d.HasChange("teams") && !d.HasChange("service_accounts") {
+		return nil
+	}
+
+	client := meta.(*client).gapi
+
+	uid := d.Get("role_uid").(string)
+	users, err := collectRoleAssignmentsToFn(d.Get("users"))
+	if err != nil {
+		return diag.Errorf("invalid user IDs specified %v", err)
+	}
+	teams, err := collectRoleAssignmentsToFn(d.Get("teams"))
+	if err != nil {
+		return diag.Errorf("invalid team IDs specified %v", err)
+	}
+	serviceAccounts, err := collectRoleAssignmentsToFn(d.Get("service_accounts"))
+	if err != nil {
+		return diag.Errorf("invalid service account IDs specified %v", err)
+	}
+
+	ra := &gapi.RoleAssignments{
+		RoleUID:         uid,
+		Users:           users,
+		Teams:           teams,
+		ServiceAccounts: serviceAccounts,
+	}
+	assignments, err := client.UpdateRoleAssignments(ra)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := setRoleAssignments(assignments, d); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
```

Comment:
Instead, we usually reuse the Read, so that we always have the same code to set attributes
```suggestion
	return ReadRoleAssignments(ctx, d meta)
```

Reply:
aeda30b61ef1b3ce244ecbfd4f5dd5c4df6749ff thnx